### PR TITLE
Frontend bitbox02 refactoring

### PR DESCRIPTION
**Pull request by thisconnect**

### CODE REVIEW

File: frontends/web/src/components/sidebar/sidebar.tsx [hunk 1]
```diff
@@ -156,7 +156,7 @@ class Sidebar extends Component<Props> {
       activeSidebar,
       sidebarStatus,
     } = this.props;
-    const hidden = ['forceHidden', 'forceCollapsed'].includes(sidebarStatus);
+    const hidden = sidebarStatus === 'forceHidden';
```
Comment (Beerosagos) [inline frontends/web/src/components/sidebar/sidebar.tsx:159]: If now there are only two possible values, that could be refactored to a bool? If it's the case, probably deserves a dedicated PR though..

**Review by Beerosagos — APPROVED** [review]
tACK, just a couple nits if you agree

File: frontends/web/src/api/devicessync.ts [hunk 2]
```diff
@@ -27,3 +28,52 @@ export const syncDeviceList = (
 ): TUnsubscribe => {
   return subscribeEndpoint('devices/registered', cb);
 };
+
+/**
+ * Fires when status of a device changed, mostly
+ * used as event to call bitbox02.getStatus(deviceID).
+ * Returns a method to unsubscribe.
+ */
+export const statusChanged = (
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/devicessync.ts:37]: I don't mind, but currently all other subscriptions use the event.data string..  

and we would have naming clash with subscription callback see `onStatusChagned` is used as statusCahanged callback:

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0995f0a550384cbed44b8b15e0459a765b152c23/frontends/web/src/routes/account/account.tsx#L128

therefore I'd prefer not to change it  in this PR. 😇

File: frontends/web/src/api/devicessync.ts [hunk 3]
```diff
@@ -27,3 +28,52 @@ export const syncDeviceList = (
...
+  deviceID: string,
+  cb: (deviceID: string) => void,
+): TUnsubscribe => {
+  const unsubscribe = subscibeLegacy('statusChanged', data => {
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/devicessync.ts:41]: 👍  makes sense, there is other subscriptions in api i.e. accountsync.ts  I'll update those in a different PR

File: frontends/web/src/components/sidebar/sidebar.tsx (hunk 1 above)
Comment (thisconnect) [reply frontends/web/src/components/sidebar/sidebar.tsx:159]: makes sense, but sidebarStatus is coming from custom store (decorator hoc) implementation which I don't want to touch and think we should remove completely.

https://github.com/digitalbitbox/bitbox-wallet-app/blob/0995f0a550384cbed44b8b15e0459a765b152c23/frontends/web/src/components/guide/guide.tsx#L37-L42

File: frontends/web/src/api/devicessync.ts [hunk 4]
```diff
@@ -15,6 +15,7 @@
  */
 
 import { subscribeEndpoint, TUnsubscribe } from './subscribe';
+import { subscribe as subscibeLegacy } from '../utils/event-legacy';
```
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/devicessync.ts]: fixed

File: frontends/web/src/api/devicessync.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/api/devicessync.ts]: renamed in https://github.com/digitalbitbox/bitbox-wallet-app/pull/2103

File: frontends/web/src/api/devicessync.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Beerosagos) [reply frontends/web/src/api/devicessync.ts:37]: I like `syncStatusChanged`! Let's keep that in mind for the next time we'll work on that :)
